src/soilbiogeochem/SoilBiogeochemNitrogenFluxType.F90

```diff
@@ -316,23 +316,22 @@ subroutine InitHistory(this, bounds)
          avgflag='A', long_name='atmospheric N deposition to soil mineral N', &
          ptr_col=this%ndep_to_sminn_col)
 
-    if ( use_fun ) then    ! redundant & confusing to write this out if FUN is on
-      this%nfix_to_sminn_col(begc:endc) = spval
-      call hist_addfld1d (fname='NFIX_TO_SMINN', units='gN/m^2/s', &
-           avgflag='A', long_name='symbiotic/asymbiotic N fixation to soil mineral N', &
-           ptr_col=this%nfix_to_sminn_col, default='inactive')
+    if (use_fun) then
+       default = 'inactive'
     else
-      this%nfix_to_sminn_col(begc:endc) = spval
-      call hist_addfld1d (fname='NFIX_TO_SMINN', units='gN/m^2/s', &
-           avgflag='A', long_name='symbiotic/asymbiotic N fixation to soil mineral N', &
-           ptr_col=this%nfix_to_sminn_col)
+       default = 'active'
     end if
+    this%nfix_to_sminn_col(begc:endc) = spval
+    call hist_addfld1d (fname='NFIX_TO_SMINN', units='gN/m^2/s', &
+         avgflag='A', long_name='symbiotic/asymbiotic N fixation to soil mineral N', &
+         ptr_col=this%nfix_to_sminn_col)
 
     this%ffix_to_sminn_col(begc:endc) = spval
     call hist_addfld1d (fname='FFIX_TO_SMINN', units='gN/m^2/s', &
          avgflag='A', long_name='free living  N fixation to soil mineral N', &
          ptr_col=this%ffix_to_sminn_col)
 
+    default = 'active'
```

Comment:
I don't think this is needed, since the `default` variable isn't used in the rest of this routine.